### PR TITLE
Feature/38 data grid component

### LINKED ISSUE #28: Data grid component
- Re-usable data grid component
- Using AgGrid with sensible defaults
- No cell renderers (for now)
- Lazy load/dynamic import AgGrid library

### PR DESCRIPTION
**Pull request by bglownia**
closes #28

### CODE REVIEW

**Comment by bglownia** [comment]
closes #77

**Review by mattrussell36 — APPROVED** [review]
Few comments here but nothing major.

I also think lets rename react-helpers to just `helpers`, `utils` or something similar

File: libs/ui-toolkit/src/components/ag-grid/ag-grid-themed.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import type { AgGridReactProps, AgReactUiProps } from 'ag-grid-react';
+import { AgGridReact } from 'ag-grid-react';
+import { ThemeContext } from '@vegaprotocol/react-helpers';
+import 'ag-grid-community/dist/styles/ag-grid.css';
+import 'ag-grid-community/dist/styles/ag-theme-balham.css';
+import 'ag-grid-community/dist/styles/ag-theme-balham-dark.css';
+
+/*
```
[the remark this answers is not in the document]
Comment (bglownia) [reply libs/ui-toolkit/src/components/ag-grid/ag-grid-themed.tsx]: I added additional layer of lazy loading to load theme specific css files. We need to have two separate implementation dynamic from 'next/dynamic'; and  React.lazy until switch to react 18

File: apps/trading/pages/_app.page.tsx [hunk 2]
```diff
@@ -25,7 +27,9 @@ function VegaTradingApp({ Component, pageProps }: AppProps) {
     []
   );
   const [dialogOpen, setDialogOpen] = useState(false);
-  const setTheme = useThemeSwitcher();
+  const [theme, setTheme] = useState<'dark' | 'light'>(getCurrentTheme());
```
[the remark this answers is not in the document]
Comment (bglownia) [reply apps/trading/pages/_app.page.tsx:30]: I did some mess in here. I've cleaned it up

File: libs/ui-toolkit/src/components/dialog/index.tsx [hunk 3]
```diff
@@ -12,8 +12,15 @@ export function Dialog({ children, open, setOpen, title }: DialogProps) {
   return (
     <DialogPrimitives.Root open={open} onOpenChange={(x) => setOpen(x)}>
       <DialogPrimitives.Portal>
-        <DialogPrimitives.Overlay className="fixed inset-0 bg-black dark:bg-white opacity-40 dark:opacity-15" />
-        <DialogPrimitives.Content className="fixed w-[500px] top-40 p-28 left-[calc(50%-250px)] dark:bg-black dark:text-white-60 bg-white text-black-60">
+        <DialogPrimitives.Overlay className="fixed inset-0 bg-black/40 dark:bg-white/15" />
+        <DialogPrimitives.Content
+          className="fixed w-[500px] p-28 dark:bg-black dark:text-white-60 bg-white text-black-60"
+          style={{
```
[the remark this answers is not in the document]
Comment (bglownia) [reply libs/ui-toolkit/src/components/dialog/index.tsx:18]: Switched to classes. inset will not work as we want

File: libs/ui-toolkit/src/components/ag-grid/ag-grid-lazy-themed.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import 'ag-grid-community/dist/styles/ag-grid.css';
+
+const AgGridLightTheme = React.lazy(() =>
+  import('./ag-grid-light').then((module) => ({
+    default: module.AgGrid,
+  }))
+);
+
+const AgGridDarkTheme = React.lazy(() =>
+  import('./ag-grid-dark').then((module) => ({
```
Comment (dexturr) [inline libs/ui-toolkit/src/components/ag-grid/ag-grid-lazy-themed.tsx:14]: does this add prefetch to the resource? I cannot remember if we need to configure something wiht webpack comments in order to force that. Would like to load in the background during idle time if possible

**Review by dexturr — APPROVED** [review]

**Pull request merged**